cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureDnsZoneService.java

```diff
@@ -0,0 +1,264 @@
+            String resourceGroup, Map<String, String> tags) {
+
+        String networkId = networkView.getNetworkId();
+        String networkResourceGroup = networkView.getResourceGroupName();
+        List<AzurePrivateDnsZoneServiceEnum> enabledPrivateEndpointServices = getEnabledPrivateEndpointServices();
+
+        boolean dnsZonesDeployed = azureClient.checkIfDnsZonesDeployed(resourceGroup, enabledPrivateEndpointServices);
+        boolean networkLinksDeployed = azureClient.checkIfNetworkLinksDeployed(resourceGroup, networkId, enabledPrivateEndpointServices);
+        String deploymentName = generateDnsZoneDeploymentName(enabledPrivateEndpointServices, DNS_ZONES);
+        String dnsZoneDeploymentId = getDnsZoneDeploymentId(resourceGroup, azureClient, deploymentName);
+
+        if (dnsZonesDeployed && networkLinksDeployed) {
+            LOGGER.debug("Dns zones ({}) and network links already deployed in resource group {}", enabledPrivateEndpointServices, resourceGroup);
+        } else if (!dnsZonesDeployed) {
+            LOGGER.debug("Dns zones are not deployed yet!");
+            String azureNetworkId = getAzureNetwork(azureClient, networkId, networkResourceGroup).id();
+            if (isRequested(dnsZoneDeploymentId)) {
+                LOGGER.debug("Dns zones ({}) already requested in resource group {}", enabledPrivateEndpointServices, resourceGroup);
+                pollForCreation(authenticatedContext, azureClient, resourceGroup, deploymentName, dnsZoneDeploymentId, null,
+                        enabledPrivateEndpointServices);
+                if (!networkLinksDeployed) {
+                    LOGGER.debug("Deploying network links that are not deployed yet!");
+                    createMissingNetworkLinks(azureClient, azureNetworkId, resourceGroup, tags, enabledPrivateEndpointServices);
+                }
+            } else {
+                try {
+                    LOGGER.debug("Dns zones ({}) are not requested yet in resource group {}", enabledPrivateEndpointServices, resourceGroup);
+                    if (isCreated(dnsZoneDeploymentId)) {
+                        LOGGER.debug("Dns zone deployment ({}) is there in database but not deployed on Azure, resetting it..", dnsZoneDeploymentId);
+                        updateDnsZone(authenticatedContext, deploymentName, dnsZoneDeploymentId, CommonStatus.REQUESTED);
+                    } else {
+                        persistDnsZone(authenticatedContext, deploymentName, dnsZoneDeploymentId);
+                    }
+
+                    createDnsZonesAndNetworkLinks(azureClient, azureNetworkId, resourceGroup, tags, enabledPrivateEndpointServices);
+                    updateDnsZone(authenticatedContext, deploymentName, dnsZoneDeploymentId, CommonStatus.CREATED);
+
+                } catch (CloudException | DataAccessException e) {
+                    LOGGER.warn("Deployment failed due to {}, setting the resource status of {} to FAILED", e.getMessage(), deploymentName);
```

Comment:
I'm still not getting it from the code why is a polling here

Reply:
Because DataAccessException comes when the unique key for a resource is violated - meaning the same DNS Zone is meant to be inserted multiple times. In that case, the second entry waits for the poller to complete the deployment of the first entry.